pkg/odo/genericclioptions/localprovider.go

```diff
@@ -34,10 +34,14 @@ func GetValidEnvInfo(cmdline cmdline.Cmdline) (*envinfo.EnvSpecificInfo, error)
 	// Check to see if the environment file exists
 	if !envInfo.Exists() {
 		//revive:disable:error-strings This is a top-level error message displayed as is to the end user
-		return nil, errors.New(`The current directory does not represent an odo component.
-To start editing your component, use "odo dev" and open this folder in your favorite IDE. Changes will be directly reflected on the cluster.
-To deploy your component to a cluster use "odo deploy".
-Or switch to directory with a component.`)
+		return nil, errors.New(`The current directory does not represent an odo component. 
+
+To get started,
+  * Create and move to a new directory, or use an existing one.
+  * Run "odo init" from the directory to initialize an odo component.
+  * Start editing the component in an IDE and run "odo dev" to see your changes get reflected on the cluster.
+
+Visit https://odo.dev for more information.`)
```

Comment:
This looks good, but the issue was opened by the user who was trying to delete from a non-component dir. The entire messaging here is targeted for a user who hasn't created a component. 

Now one could say that the first point does tell the user to move to a directory. But it's not accurate.

Maybe we shouldn't display a generic message for `odo init` and `odo delete`. We could tailor a message specific to each one of them.

Reply:
>Maybe we shouldn't display a generic message for odo init and odo delete. We could tailor a message specific to each one of them.

That was my initial idea. But then I had to figure out how to determine if the directory was empty, and changing the error message seemed like simplest thing to do. I'll get to it.

Reply:
[39758f0](https://github.com/redhat-developer/odo/pull/6006/commits/39758f0e45d3b2544a136d3b2aab01cc29b7d8e6) feels like over engineering, but eh :shrug: